chichiwang/exercise-js
# Build Home page section: Tech Stack

## Description
Linked to Issue: #64
Build Home page section "Tech Stack". Update the `README.md` section "Tech Stack" to include the correct technologies.

Create a component `pages/components/ExternalLink` to make setting up an anchor tag to an external site easier:
* Applies `rel` attribute with `noreferrer` value
* Applies `target` attribute with `_blank` value

Use this `ExternalLink` component for the Github project link in the `Nav` component, as well as for all external links in the `Home` page component.

Update the `HeadingBlock` component to accept a `className` prop. Update the `TaipeiBuilding` component to remove the sign on the building.

## Changes
* Update "Technology Stack" section of `Readme.md`
  * Rename section to "Tech Stack"
  * Add `React`, `React Router`, and `Pino` to Server stack
  * Add `React Router` to Client stack
  * Add `React Testing Library` to Developer Tooling stack
* Update `pages/Home/index.tsx` component
  * Replace anchor tags to external sites with new `ExternalLink` component
  * Create a "Tech Stack" section
* Update the `pages/Layout/Nav.tsx` component
  * Replace the anchor tag to Github project with new `ExternalLink` component
* Create a `pages/components/ExternalLink` component
  * Renders an anchor tag with `rel` attribute set to `noreferrer` and `target` attribute set to `_blank`
* Update `pages/components/HeadingBlock/index.tsx` component to accept a `className` attribute
* Update `pages/components/HeroBanner/SceneRed/TaipeiBuilding.tsx` component to remove the sign from the building

## Steps to QA
* Pull down and switch to this branch
* Run this branch locally and verify in browser:
  * The Taipei 101 building in the hero banner does not have the logo on it
  * The Github project link in the header still opens this repository in a new tab
  * The Home page looks correct, and the "Tech Stack" section looks good in all screen resolutions
  * The print preview for the Home page does not break the homepage in an awkward place between pages
